drivers/python/rethinkdb/net.py

```diff
@@ -515,6 +523,12 @@ def __init__(self, conn_type, host, port, db, auth_key, timeout, ssl, **kwargs):
         self._instance = None
         self._next_token = 0
 
+    def client_port(self):
+        return self._instance._socket.client_port()
```

Comment:
@larkost latest commit addresses the possible null reference you pointed out. All the tests now ensure that these two methods return `None` when called on a closed connection.